pkg/lib/authenticationflow/declarative/intent_bot_protection.go

```diff
@@ -0,0 +1,57 @@
+package declarative
+
+import (
+	"context"
+
+	"github.com/iawaknahc/jsonschema/pkg/jsonpointer"
+
+	authflow "github.com/authgear/authgear-server/pkg/lib/authenticationflow"
+	"github.com/authgear/authgear-server/pkg/lib/feature/botprotection"
+)
+
+func init() {
+	authflow.RegisterIntent(&IntentBotProtection{})
+}
+
+type IntentBotProtection struct {
+	OneOfJSONPointer jsonpointer.T `json:"json_pointer,omitempty"`
+}
+
+var _ authflow.Intent = &IntentBotProtection{}
+
+func (*IntentBotProtection) Kind() string {
+	return "IntentBotProtection"
+}
+
+func (i *IntentBotProtection) CanReactTo(ctx context.Context, deps *authflow.Dependencies, flows authflow.Flows) (authflow.InputSchema, error) {
+	// The last node is NodeDidVerifyBotProtection
+	// So if MilestoneDidVerifyBotProtection is found, this intent has finished
+	_, _, ok := authflow.FindMilestoneInCurrentFlow[MilestoneDidVerifyBotProtection](flows)
+	if ok {
+		return nil, authflow.ErrEOF
+	}
+
+	return &InputSchemaBotProtectionVerification{
+		JSONPointer: i.OneOfJSONPointer,
+	}, nil
+}
+
+func (i *IntentBotProtection) ReactTo(ctx context.Context, deps *authflow.Dependencies, flows authflow.Flows, input authflow.Input) (*authflow.Node, error) {
+	var inputBotProtectionVerification inputBotProtectionVerification
+	if !authflow.AsInput(input, &inputBotProtectionVerification) {
+		return nil, authflow.ErrIncompatibleInput
+	}
+
+	err := deps.BotProtection.Verify(inputBotProtectionVerification.GetBotProtectionType(), inputBotProtectionVerification.GetBotProtectionResponse())
+
+	switch err {
```

Comment:
First of all, you cannot switch err in this way. Error is Go is a chain of errors.

The standard way to handle error in Go is to use errors.Is and errors.As. Both of them return bool so you can do

```golang
err := MyFunction()
var errB *ErrorB
switch { // This is a switch without subject. This is a valid usage of switch in Go. See https://go.dev/tour/flowcontrol/11
case errors.Is(err, ErrorConstantA):
  break
case errors.Is(err, &errB):
  break
}
```